core/src/main/java/hudson/FilePath.java

```diff
@@ -564,39 +564,38 @@ public Void invoke(File dir, VirtualChannel channel) throws IOException, Interru
      *      Compression mode of this tar file.
      * @since 1.292
      * @see #untarFrom(InputStream, TarCompression)
+     * @return
      */
-    public void untar(final FilePath target, final TarCompression compression) throws IOException, InterruptedException {
+    public List<File> untar(final FilePath target, final TarCompression compression) throws IOException, InterruptedException {
```

Comment:
that's a public api change. not sure we want that.
But this case changing void to a return type should be ok but still if someone use some reflection it will be a problem. Maybe a new method called untar2 or untarFiles?

Reply:
I thought about conflicts, more likely with SpotBugs, but if reflection is an option, is better to overload the method.